R/new.R

```diff
@@ -114,13 +114,7 @@ validate_nrow <- function(names, lengths, nrow) {
 }
 
 update_tibble_attrs <- function(x, ...) {
-  # Compat for https://github.com/r-spatial/sf/pull/1232
-  attribs <- list(...)
-  if (has_length(attribs) && is_named(attribs)) {
-    attributes(x)[names(attribs)] <- attribs
-  }
-
-  x
+  .Call(`tibble_update_attrs`, x, pairlist2(...))
```

Comment:
Probably safer to check if any of the special attributes is supplied, and fall back to the R code in that case?

Reply:
I'm not sure how safer that would be, `setAttrib()` does some internal dispatch to deal with special attributes, i.e. `names`, `row.names`, `class` ...

Reply:
oh yes, I was confused!

There's still some special handling at R level in `structure()` though, but that doesn't seem important:

```r
specials <- c(".Dim", ".Dimnames", ".Names", ".Tsp", ".Label")
```